pointcloud/basic_octree_utils.go

```diff
@@ -165,11 +161,25 @@ func (octree *basicOctree) helperIterate(lowerBound, upperBound, idx int, fn fun
 			idx += numPoints
 		}
 
-	case LeafNodeFilled:
+	case leafNodeFilled:
 		ok = fn(octree.node.point.P, octree.node.point.D)
 
-	case LeafNodeEmpty:
+	case leafNodeEmpty:
 	}
 
 	return ok
 }
+
+// Helper function for calculating the center of a pointcloud based on its metadata.
+func getCenterFromPcMetaData(meta MetaData) r3.Vector {
+	return r3.Vector{
+		X: meta.MinX + (meta.MaxX-meta.MinX)/2,
+		Y: meta.MinY + (meta.MaxY-meta.MinY)/2,
+		Z: meta.MinZ + (meta.MaxZ-meta.MinZ)/2,
```

Comment:
[opt] This can also be expressed as `(meta.MinX + meta.MaxX)/2` but it's not like an extra operation makes a big difference.